Libplanet/Net/Messages/RecentStates.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Security.Cryptography;
+using NetMQ;
+
+namespace Libplanet.Net.Messages
+{
+    internal class RecentStates : Message
+    {
+        public RecentStates(
+            HashDigest<SHA256> blockHash,
+            IImmutableDictionary<
+                HashDigest<SHA256>,
+                IImmutableDictionary<Address, object>
+            > blockStates,
+            IImmutableDictionary<Address, IImmutableList<HashDigest<SHA256>>> stateReferences,
+            IImmutableDictionary<Address, long> txNonces
+        )
+        {
+            BlockHash = blockHash;
+
+            if (blockStates is null && stateReferences is null && txNonces is null)
+            {
+                BlockStates = null;
+                StateReferences = null;
+                TxNonces = null;
+                return;
+            }
+
+            if (blockStates is null)
+            {
+                throw new ArgumentNullException(nameof(blockStates));
+            }
+            else if (stateReferences is null)
```

Comment:
It isn't really necessary, but I intended to use `else if` here instead of `if` because lint warns that I have to insert a new line above every `if` statement.  What's your opinion?

Reply:
Ok, I got it. I prefer `if` with a new line, but it depends on your taste.